components/viz/service/display/software_output_device.h

```diff
@@ -83,6 +83,11 @@ class VIZ_SERVICE_EXPORT SoftwareOutputDevice {
   // platform's proposed surface size.
   virtual bool SupportsOverridePlatformSize() const;
 
+  // [RUN-2989] Allow Replay commmand handlers to read the viewport size.
+  const gfx::Size& ReplayViewportPixelSize() const {
+    return viewport_pixel_size_;
```

Comment:
mind making this function return `gfx::Size*` as well and returning `&viewport_pixel_size_` here (and removing the  `&` from `GetCurrentViewportPixelSizeImpl`)?

the current formulation of things makes `GetCurrentViewportPixelSizeImpl` look like it's opening us up to a use-after-free.